src/snowflake/snowpark/modin/plugin/extensions/datetime_index.py

```diff
@@ -1533,20 +1535,26 @@ def mean(
         >>> idx = pd.date_range('2001-01-01 00:00', periods=3)
         >>> idx
         DatetimeIndex(['2001-01-01', '2001-01-02', '2001-01-03'], dtype='datetime64[ns]', freq=None)
-        >>> idx.mean()  # doctest: +SKIP
+        >>> idx.mean()
         Timestamp('2001-01-02 00:00:00')
         """
+        # Need to convert timestamp to int value (nanoseconds) before aggregating.
+        # TODO: SNOW-1625233 When `tz` is supported, add a `tz` parameter to `to_datetime` for correct timezone result.
```

Comment:
can you raise a NotImplementedError for tz?

Reply:
I don't think that's possible since `tz` itself is not supported for DatetimeIndex